### ISSUE TITLE
Big files crash the Python client

### ISSUE THREAD

**Issue by vmarkovtsev** [src-d/lookout#143]
Python grpc implementation has the hardcoded received message size limit which is 4MB.
Actually, it is "hardcoded" and there is an undocumented working hacky way to remove the limit:

```python
channel = grpc.insecure_channel("address:port", options=[
                    (grpc._cython.cygrpc.ChannelArgKey.max_send_message_length, -1),
                    (grpc._cython.cygrpc.ChannelArgKey.max_receive_message_length, -1)])
```

It is [very much discouraged to do this](https://github.com/grpc/grpc/issues/7927#issuecomment-244845823) and the grpc devs promise an "incorrect behavior".

We do hit this limit on Python side while streaming `GetChanges`, one of the files appears big enough to explode with a huge UAST:

```
[2018-08-15T16:49:31.72710336+02:00]  INFO streaming server call finished app=lookout duration=1.341349114s error=rpc error: code = Canceled desc = context canceled grpc.code=Canceled grpc.method=GetFiles grpc.service=pb.Data grpc.start_time=2018-08-15T16:49:30+02:00 source=grpchelper/logger.go:19 span.kind=server system=grpc
[2018-08-15T16:49:31.728291173+02:00]  INFO streaming client call finished app=lookout duration=1.365006395s error=rpc error: code = Internal desc = <class 'grpc._channel._Rendezvous'>: <_Rendezvous of RPC that terminated with:
	status = StatusCode.RESOURCE_EXHAUSTED
	details = "Received message larger than max (6584089 vs. 4194304)"
	debug_error_string = "{"created":"@1534344571.726949437","description":"Error received from peer","file":"src/core/lib/surface/call.cc","file_line":1083,"grpc_message":"Received message larger than max (6584089 vs. 4194304)","grpc_status":8}"
> grpc.code=Internal grpc.method=NotifyReviewEvent grpc.service=pb.Analyzer grpc.start_time=2018-08-15T16:49:30+02:00 source=grpchelper/logger.go:19 span.kind=client system=grpc
[2018-08-15T16:49:31.728388447+02:00] ERROR analysis failed analyzer=test-analyzes app=lookout error=rpc error: code = Internal desc = <class 'grpc._channel._Rendezvous'>: <_Rendezvous of RPC that terminated with:
	status = StatusCode.RESOURCE_EXHAUSTED
	details = "Received message larger than max (6584089 vs. 4194304)"
	debug_error_string = "{"created":"@1534344571.726949437","description":"Error received from peer","file":"src/core/lib/surface/call.cc","file_line":1083,"grpc_message":"Received message larger than max (6584089 vs. 4194304)","grpc_status":8}"
> head=HEAD provider= repository=file:///home/vadim/Projects/style-analyzer source=server/server.go:256
```

We need to do something better than the workaround. Solutions:

1. Drop the big files.
2. Stream file UAST (the text always fits into 4MB, it is the UAST which explodes). This will be a pain in the ass for those who send and for those who receive.
3. Compress: lz4, gzip, lzma. In my experience, lzma reduces the UAST size x10.

**Comment by vmarkovtsev** [comment]
P.S.: Python grpc never stops to suck.

**Comment by vmarkovtsev** [comment]
@bzz This snippet segfaults the Python interpreter, checked it yesterday.

**Comment by bzz** [comment]
Thee one that I have posted today? :) Wow! 

Ok, as it seems to work for other people - there might be a bug on how those limits are set on any of the sides, so before anything we should verify that.

**Comment by vmarkovtsev** [comment]
@bzz I got it, when I googled I saw a slightly different snippet:

```python
channel = grpc.insecure_channel(hostname + ':' + str(port), options={
          'grpc.max_send_message_length': 50 * 1024 * 1024,
          'grpc.max_receive_message_length': 50 * 1024 * 1024,
})
```

and that indeed segfaults. Passing the options as a list of tuples does not crash. I still have a weird behavior but that 's another issue.

**Comment by vmarkovtsev** [comment]
Related to src-d/style-analyzer#36

**Comment by vmarkovtsev** [comment]
I do not confirm that it works though because is blocked by #144

**Comment by bzz** [comment]
Awesome 🎉
